File: src/#203/solution.js
function ListNode(val, next) {
  this.val = val === undefined ? 0 : val
  this.next = next === undefined ? null : next
}

var removeElements = function (head, val) {
  let dummy = new ListNode(-1, head)
  let p = dummy
  while (p.next !== null) {
    if (p.next.val == val) {
      p.next = p.next.next
      continue
    }
    p = p.next
  }
  return dummy.next
}

File: src/#203/solution.go
package solution

type ListNode struct {
	Val  int
	Next *ListNode
}

func removeElements(head *ListNode, val int) *ListNode {
	dummy := &ListNode{-1, head}
	p := dummy
	for p.Next != nil {
		if p.Next.Val == val {
			p.Next = p.Next.Next
			continue
		}
		p = p.Next
	}
	return dummy.Next
}
